src/Xamarin.Android.Build.Tasks/Utilities/NativeAssemblyGenerator/NativeAssemblyGenerator.cs

```diff
@@ -0,0 +1,1223 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Text;
+
+using Xamarin.Android.Tools;
+
+namespace Xamarin.Android.Tasks
+{
+	abstract class NativeAssemblyGenerator
+	{
+		protected const string AutomaticStringSection = "autostr";
+
+		internal class NativeType
```

Comment:
Given that the declaring class is `internal`, do these & other types need to *also* be `internal`, or should they just be `public`?